numpy_financial/tests/test_financial.py

```diff
@@ -421,20 +421,20 @@ def test_rate_nan(self, number_type):
         Both payments and present value are positive, it is impossible to pay off the existing balance
         by making further withdrawals, regardless of the rate.
         """
-        rate = numpy.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(0))
+        rate = npf.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(0))
         assert_equal(numpy.nan, float(rate))
-        rate = numpy.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(5000))
+        rate = npf.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(5000))
         assert_equal(numpy.nan, float(rate))
 
         # begin
-        rate = numpy.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 1)
+        rate = npf.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 1)
         assert_equal(numpy.nan, float(rate))
-        rate = numpy.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 'begin')
+        rate = npf.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 'begin')
         assert_equal(numpy.nan, float(rate))
```

Comment:
Another port from the original PR:

> I think these tests could be stronger. The tests should ensure rate propagates the required type, so Decimal inputs return Decimal values. -- @Kai-Striega

Reply:
@Kai-Striega  test_financial.py is having test_rate_decimal() function which test the scenario in which input is decimal and output is also decimal. I can add few more tests, If you think we need some more scenarios based on the decimal input and output.

Reply:
@Mandeep0001 test_rate_decimal tests the types when ``rate(...)`` returns a valid answer. In this case a rate doesn't exist that satisfies the conditions so `nan` is returned. However `np.nan` and `decimal.Decimal('nan')` are not the same, they are different types.

Your solution (``default_type(np.nan) + rn``) handles this well. It returns the correct value and propagates the users' type. Well done. I think the tests could be stronger as the type is not being tested, for example ``np.nan + np.empty_like(rn, dtype=float)`` would pass the test, despite returning a np.float rather than a decimal.Decimal.